webapp/channels/src/components/sidebar/sidebar_category/sidebar_category_menu/create_new_category_menu_item.tsx

```diff
@@ -19,6 +19,7 @@ type Props = {
 
 const CreateNewCategoryMenuItem = ({
     id,
+    ...otherProps
```

Comment:
If we are accepting other Props, the Props type for this component should take account of them.

Reply:
It's not so much that we're accepting extra props into here but rather that we're transparently passing along props provided by MUI. I don't think we should add them to `Props` since that might make people think they should provide them themselves, and it may cause TS to complain because of that as well

Reply:
But those props have to be received and send, and be "somewhere". My approach would be something like:

```
type Props = {
    myProp1,
    myProp2,
} & MUIProps
```

With that and the fact the body does something like...
```
const Foo = ({myProp1, myProp2, ...otherProps}) =>
```

Provides clarity towards what the component uses. On top of that, any caller knows that it has to be used as a MUI component, with any other prop that MUI can use or needs to use.

If there is something that we are "fixing", we can do something like...

```
type Props = {
    myProp1,
    myProp2,
} & Omit<MUIProps, 'nonConfigurableProp1'|'nonConfigurableProp2'>
```